sky/utils/cli_utils/status_utils.py

```diff
@@ -138,7 +139,11 @@ def show_cost_report_table(cluster_records: List[_ClusterCostReportRecord],
             columns.append(status_column.name)
     cluster_table = log_utils.create_table(columns)
 
-    for record in cluster_records:
+    num_lines_to_display = _NUM_COST_REPORT_LINES
+    if show_all:
+        num_lines_to_display = len(cluster_records)
+
+    for record in cluster_records[:num_lines_to_display]:
```

Comment:
The cost records seems to be in the order of launch time, but it might be better to place the clusters that have not been terminated yet at the beginning.